fineract-core/src/main/java/org/apache/fineract/infrastructure/core/domain/BatchRequestContextHolder.java

```diff
@@ -18,25 +18,20 @@
  */
 package org.apache.fineract.infrastructure.core.domain;
 
+import com.alibaba.ttl.TransmittableThreadLocal;
 import java.util.Map;
 import java.util.Optional;
-import org.springframework.core.NamedThreadLocal;
 import org.springframework.transaction.TransactionStatus;
 
 public final class BatchRequestContextHolder {
 
     private BatchRequestContextHolder() {}
 
-    private static final ThreadLocal<Map<String, Object>> batchAttributes = new NamedThreadLocal<>("batchAttributes");
+    private static final ThreadLocal<Map<String, Object>> batchAttributes = new TransmittableThreadLocal<>();
 
-    private static final ThreadLocal<Optional<TransactionStatus>> batchTransaction = new NamedThreadLocal<>("batchTransaction") {
+    private static final ThreadLocal<Optional<TransactionStatus>> batchTransaction = TransmittableThreadLocal.withInitial(Optional::empty);
```

Comment:
Do we need thread local value copies to child threads?

Reply:
Later for the "async" mode yes. See my previous comments for the dependency. I think I'll remove for now to keep this as small as possible and avoid distractions. But later we have to find a solution here if we want to use the full performance potential.